source/opt/def_use_manager.cpp

```diff
@@ -121,11 +120,12 @@ bool DefUseManager::ReplaceAllUsesWith(uint32_t before, uint32_t after) {
       if (it->inst->type_id() != 0 && it->operand_index == 0) {
         it->inst->SetResultType(after);
       } else if (it->inst->type_id() == 0) {
-        assert(false &&
-               "Result type id considered as using while the instruction "
-               "doesn't have a result type id.");
+        SPIRV_ASSERT(consumer_, false,
+                     "Result type id considered as use while the instruction "
+                     "doesn't have a result type id.");
```

Comment:
Good catch! Done in the second commit. But I used `std::exit`, which does the "all C streams are flushed and closed" cleanup.